src/filesystem/unix/SDL_sysfilesystem.c

```diff
@@ -594,6 +595,16 @@ char *SDL_GetUserFolder(SDL_Folder folder)
         return NULL;
     }
 
+    newretval = (char *) SDL_realloc(retval, SDL_strlen(retval) + 2);
```

Comment:
Maybe this logic should be moved into xdg_user_dir_lookup()?

Reply:
I did think about that, but when I tried, I found that the code would need to be duplicated at two places (the two non-null returns). It would also modify the structure of the function, which would make it more difficult to incorporate upstream changes, if any.

Should I give it a try anyways?

Reply:
No, that was the right call.